polygeist/lib/Conversion/PolygeistToLLVM/PolygeistToLLVM.cpp

```diff
@@ -1524,7 +1539,13 @@ struct ConvertPolygeistToLLVMPass
       populateCallOpTypeConversionPattern(patterns, converter);
       populateAnyFunctionOpInterfaceTypeConversionPattern(patterns, converter);
 
+      constexpr auto clientAPI = spirv::ClientAPI::OpenCL;
+
+      populateSPIRVToLLVMConversionPatterns(converter, patterns, clientAPI);
```

Comment:
Wondering when we should add new passes to the pipeline and when we should add patterns to an existing pass.
Maybe we should break PolygeistToLLVM pass into passes, so optimizations can operate in between? (It can be address in future PR.)

Reply:
Yes, that's something we should work on... This pass just does too many things.